R/utilities-file.R

```diff
@@ -79,3 +79,65 @@ readExcel <- function(path, sheet = NULL, ...) {
   # Write the data into an excel file.
   writexl::write_xlsx(data, path = path, col_names = col_names)
 }
+
+#' Write data to excel with append capability
+#'
+#' @details Uses `openxlsx` to write data to excel with ability to append to 
+#' existing files. If the folder does not exist, creates folder(s) recursively.
+#' If the file exists, can append new sheets or append data to existing sheets.
+#'
+#' @param data Data frame or named list of data frames that will be sheets in
+#' the xlsx
+#' @param path Path to the xlsx file
+#' @param append If TRUE and file exists, appends data. If FALSE (default), 
+#' overwrites the file.
+#' @param col_names Whether to include column names
+#' @keywords internal
+#' @noRd
+.writeExcelWithAppend <- function(data, path, append = FALSE, col_names = TRUE) {
```

Comment:
Why not simply extend the existing `.writeExcel` function by an `append` argument? I am not in favor of having these two functions simultaneously now